src/rproc/python/schema_migration.py

```diff
@@ -0,0 +1,78 @@
+# This file is part of qserv.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+
+"""Module defining methods used in schema migration of rproc database.
+"""
+
+__all__ = ["make_migration_manager"]
+
+import logging
+
+from lsst.qserv.schema import SchemaMigMgr, Uninitialized
+
+
+_log = logging.getLogger(__name__)
+qservResultDb = "qservResult"
+
+
+class RprocMigrationManager(SchemaMigMgr):
```

Comment:
I'm still seeing these classes in Qserv modules. Each does nearly exactly the same. At one point I heard a comment from @andy-slac that we need to have a generic class for that. And for those cases when the standard behavior of that class wouldn't be suitable (non-trivial data migration required by the new schema) then we should have an option to set up an overload.
I'm not sure what would be the best way to implement this in Python.

Reply:
In python, inheritance is specified in the parens after the class, so `RprocMigrationManager` inherits from `SchemaMigMgr`. The common code is in the latter class as of a year or so ago IIRC.

Reply:
I still don't understand why you need a special class for each nodule even if all module-specific classes essentially do the same. The only difference between them is that they print slightly different messages. I can imagine having the schema migration tool which would simply look at the `schema/<module>/` folder to locate schema versions and do the job. And if some module has a special procedure for migrating its schema then perhaps some optional module-specific class would be pulled and used. Otherwise, having these (essentially) empty classes make no sense to me.

Reply:
Currently there's no requirement that modules manage their schema version the same way, subclassing allows specialization of getting & setting the version.